pkg/generate/code/set_sdk.go

```diff
@@ -718,6 +722,139 @@ func SetSDKSetAttributes(
+// setSDKReadMany is a special-case handling of those APIs where there is no
+// ReadOne operation and instead the only way to grab information for a single
+// object is to call the ReadMany/List operation with one of more filtering
+// fields-- specifically identifier(s). This method populates this identifier
+// field with the identifier shared between the shape and the CR. Note, in the
+// case of multiple matching identifiers, the identifier field containing 'Id'
+// will be the only field populated.
+//
+// As an example, DescribeVpcs EC2 API call doesn't have a ReadOne operation or
+// required fields. However, the input shape VpcIds field can be populated using
+// a VpcId, a field in the VPC CR's Status. Therefore, populate VpcIds field
+// with the *single* VpcId value to ensure the returned array from the API call
+// consists only of the desired Vpc.
+func setSDKReadMany(
+	cfg *ackgenconfig.Config,
+	r *model.CRD,
+	op *awssdkmodel.Operation,
+	sourceVarName string,
+	targetVarName string,
+	indentLevel int,
+) string {
+	inputShape := op.InputRef.Shape
+	if inputShape == nil {
+		return ""
+	}
+
+	out := "\n"
+	indent := strings.Repeat("\t", indentLevel)
+
+	resVarPath := ""
+	pluralize := pluralize.NewClient()
+	opConfig, override := cfg.OverrideValues(op.Name)
+	shapeIdentifiers := FindIdentifiersInShape(r, inputShape)
+	var err error
+	for memberIndex, memberName := range inputShape.MemberNames() {
+		if override {
+			value, ok := opConfig[memberName]
+			memberShapeRef, _ := inputShape.MemberRefs[memberName]
+			memberShape := memberShapeRef.Shape
+			if ok {
+				switch memberShape.Type {
+				case "boolean", "integer":
+				case "string":
+					value = "\"" + value + "\""
+				default:
+					panic("Member type not handled")
+				}
+
+				out += fmt.Sprintf("%s%s.Set%s(%s)\n", indent, targetVarName, memberName, value)
+				continue
+			}
+		}
+
+		// Field renames are handled in getSanitizedMemberPath
+		resVarPath, err = getSanitizedMemberPath(memberName, r, op, sourceVarName)
+		if err != nil {
+			// if it's an identifier field check for singular/plural
+			if util.InStrings(memberName, shapeIdentifiers) {
+				var flipped string
+				if pluralize.IsPlural(memberName) {
+					flipped = pluralize.Singular(memberName)
+				} else {
+					flipped = pluralize.Plural(memberName)
+				}
+				// If there are multiple identifiers, then prioritize the
+				// 'Id' identifier.
+				if resVarPath == "" || (!strings.HasSuffix(resVarPath, "Id") ||
+					!strings.HasSuffix(resVarPath, "Ids")) {
+					resVarPath, err = getSanitizedMemberPath(flipped, r, op, sourceVarName)
+				}
+			} else {
+				// TODO(jaypipes): check generator config for exceptions?
+				continue
+			}
+		}
+
+		memberShapeRef, _ := inputShape.MemberRefs[memberName]
+		memberShape := memberShapeRef.Shape
+		out += fmt.Sprintf(
+			"%sif %s != nil {\n", indent, resVarPath,
+		)
+
+		switch memberShape.Type {
+		case "list":
+			// Expecting slice of identifiers
+			{
```

Comment:
nit: I believe that creating a new scope is not necessary here

Reply:
I think Bryan is just copying code from other places in this file. :)

Reply:
^^ I stand on the shoulder's of giants